### ISSUE TITLE
[Idea] "Archive" done tasks

### ISSUE THREAD

**Issue by diego898** [NicolasPetton/Projectify#70]
It would be a great feature to allow you "archive" done tasks - as in, "compress" them all into a bulletted list, or data tiddler, etc. and delete their actual full tiddlers. 

I have quite a few ephemeral tiddlers lying around that will never be resurrected, and so don't need to be coming up in searches, tags, etc.

**Comment by diego898** [comment]
Here is a simple "Archive" button that places as all todos tagged with done in an Archive tiddler as JSON.

Note: multiple rounds of archiving will just keep appending the export to the bottom, meaning the import process/un-archiving would be more complicated, but this is just a proof of concept 


```
<$vars archiveTiddlerTitle="Archive" 
       todoText=<<jsontiddlers "[tag[done]]" >> 
       emptyJSON="[]" linebreak="

"> 
    <$set name="newArchiveText" 
          filter="[<todoText>!match<emptyJSON>]" 
          value={{{ [title<archiveTiddlerTitle>get[text]addsuffix<linebreak>addsuffix<todoText>] }}} 
          emptyValue={{{ [title<archiveTiddlerTitle>get[text]] }}} >

        <$button>
            <$action-setfield $tiddler=<<archiveTiddlerTitle>> 
                              $field="text" 
                              $value=<<newArchiveText>> />
            <$action-deletetiddler $filter="[tag[done]]" />           
            Archive! 
        </$button>

    </$set>
</$vars>
```
```

**Comment by pmario** [comment]
Good idea. ... Should it be possible to "resurrect" ToDos from the archive? ... just curious.

**Comment by diego898** [comment]
@pmario - definitely. I posted about this on the GG as well. 

I think a problem towards un-archiving is:

*  In the proof-of-concept solution I gave above, different archiving batches will just be appended as separate JSON objects, instead of merging everything back into one large JSON object.
* Even if that was solved, actually building the unarchiver is probably beyond my abilities, though it might be a good use case for Joshua's JSON Mangler.
